Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,443 @@
+                $codespace.owner.login | Should -Be $script:OwnerName
+            }
+        }
+    }
+
+    Context 'When getting a codespace for a specified codespace name' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace
+            $codespace = Get-GitHubCodespace -CodespaceName $codespaces[0].name
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+        }
+    }
+
+    Context 'When specifiying the Uri parameter' {
+        BeforeAll {
+            $codespace = Get-GitHubCodespace -Uri $repo.RepositoryUrl
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+
+    Context "When specifiying the Uri parameter from the pipeline" {
+        BeforeAll {
+            $codespace = $repo | Get-GitHubCodespace
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+}
+
+
+Describe 'GitHubCodespaces\New-GitHubCodespace' {
+    Context -Name 'When creating a repository for the authenticated user' {
+        Context -Name 'When creating a codespace with default settings with RepositoryId' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    RepositoryId = $repo.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.owner.UserName | Should -Be $script:OwnerName
+                $codespace.template | Should -BeNullOrEmpty
+            }
+
+            AfterAll {
+                if ($codespace)
+                {
+                    Remove-GitHubCodespace -CodespaceName $codespace.name -Confirm:$false
+                }
+            }
+        }
+
+        Context -Name 'When creating a codespace with default settings with Ref' {
+            BeforeAll {
+                $repoWithPR = Get-GitHubRepository -OrganizationName $script:organizationName |
```

Comment:
This one is going to fail when run against the default test accounts, as they have no repos in them when they run (unless there were failures in previous runs resulting in an incomplete cleanup, and then even if they did, no guarantee that they have any PR's).  You can make it a bit more robust by ensuring that there is a specific repo in there with a known PR:

1. Creating a new repo (`New-GitHubRepository`)
2. Creating a new branch in that repo (`New-GitHubRepositoryBranch`)
3. Making a change to the test branch (`Set-GitHubContent`)
4. Creating a new PR for that branch against the main branch (`New-GitHubPullRequest`)

```powershell
$mainBranchName = 'master'
$prBranchName = 'test'
$repoWithPR = New-GitHubRepository -RepositoryName ([guid]::NewGuid()) -AutoInit
$prBranch = $repoWithPR | New-GitHubRepositoryBranch -BranchName $mainBranchName -TargetBranchName $prBranchName
$repoWithPR | Set-GitHubContent -Path 'README.md' -CommitMessage 'test' -Content 'New content' -BranchName $prBranchName
$pullRequest = $branch | New-GitHubPullRequest -Title 'testPR' -Head $prBranchName -Base $mainBranchName -Verbose
```

> Just be sure to clean it up (`Remove-GitHubRepository`) when you're all done with it.

Reply:
Common Before/After All creates and destroys a $repo, so I've reused that but did create a new branch and PR as suggested.

My default test org's default branch is main, so I opted for dynamicly retrieving with `$mainBranchName = $repo | Get-GitHubRepositoryBranch | Select-Object -ExpandProperty name -First 1`.  Let me know if there's a better way.